RunCommand.lua

```diff
@@ -1,104 +1,138 @@
+local runScriptFromSelectionButton: PluginToolbarButton = toolbar:CreateButton("Run From Selected", "Create an RunCommand", "rbxassetid://14978048121")
+local runScriptFromEditorButton: PluginToolbarButton = toolbar:CreateButton("Run From Editor", "Create an RunCommand", "rbxassetid://14978048121")
 
-runScriptButton.ClickableWhenViewportHidden = true
 openScriptButton.ClickableWhenViewportHidden = true
-runScriptButton.Enabled = false
-
-local function GetRunCommandFolder()
-	local runCommandFolder = game:GetService("ServerScriptService"):FindFirstChild("RunCommands")
-	if not runCommandFolder then
-		runCommandFolder = Instance.new("Folder")
-		runCommandFolder.Name = "RunCommands"
-		runCommandFolder.Parent = game:GetService("ServerScriptService")
-	end
+runScriptFromSelectionButton.ClickableWhenViewportHidden = true
+runScriptFromEditorButton.ClickableWhenViewportHidden = true
+
+runScriptFromSelectionButton.Enabled = false
+runScriptFromEditorButton.Enabled = false
+
+-- Creates a folder or fetches the current one 
+local function GetRunCommandFolder(): Folder
+
+	local runCommandFolder: Folder = ServerScriptService:FindFirstChild("RunCommands") or Instance.new("Folder", ServerScriptService)
+	runCommandFolder.Name = "RunCommands"
+
 	return runCommandFolder
+
 end
 
+-- Executes the script that is given to the function
 local function ExecuteScript(selectedScript: Script)
-	local runCommandFolder = GetRunCommandFolder()
-	local newScript = Instance.new("ModuleScript")
+
+	local newScript: ModuleScript = Instance.new("ModuleScript")
+
 	newScript.Name = HttpService:GenerateGUID()
-	
-	local wrapperCode = `\
-		local thread = coroutine.create(function() {selectedScript.Source} end)\
-		return thread\
-	`
-	ScriptEditorService:UpdateSourceAsync(newScript, function(oldContent)
+
+	local wrapperCode = `\n return coroutine.create(function() {selectedScript.Source} end)`
+	ScriptEditorService:UpdateSourceAsync(newScript, function(oldContent: string)
 		return wrapperCode
 	end)
 
-	local thread: thread
-	local destroyListener = selectedScript:GetPropertyChangedSignal("Parent"):Connect(function()
-		print("Destroying Module Script")
+	local thread: thread = require(newScript) :: thread
+	local destroyListener: RBXScriptConnection? = selectedScript:GetPropertyChangedSignal("Parent"):Connect(function()
 		if thread then
+			print("Destroying Module Script")
 			coroutine.close(thread)
 		end
 	end)
-	
-	local success, runtimeError = pcall(function()
-		thread = require(newScript)
-	end)
-	coroutine.resume(thread)
+
+	local success: boolean, runtimeErrorMessage: string = coroutine.resume(thread)
+
+	if not success then
+		warn(selectedScript.Name, " got this error: ")
+		warn(runtimeErrorMessage)
+	end
+
 	while coroutine.status(thread) ~= "dead" do
 		task.wait(1)
 	end
+
 	if destroyListener then
 		destroyListener:Disconnect()
 		destroyListener = nil
 	end
+
```

Comment:
Tons of new lines?

Reply:
Not intentional at all, you can remove it. I probably was editing and forgot to delete the dead space.